src/indicators/drought/biomasse.py

```diff
@@ -0,0 +1,335 @@
+
+
+def download_dmp(admin: AdminArgument = "ADM2"):
+    """Download raw DMP data
+
+    Raw DMP data is downloaded for specified
+    administrative area. The data is downloaded
+    in CSV format and can be processed using
+    ``process_dmp()``.
+    """
+    _check_admin(admin)
+    url = _base_url.format(admin=admin)
+    raw_path = _get_raw_path(admin)
+    download_ftp(url=url, save_path=raw_path)
+    return None
+
+
+def load_dmp(admin: AdminArgument = "ADM2", redownload: bool = False):
+    _check_admin(admin)
+    raw_path = _get_raw_path(admin)
+    if not os.path.exists(raw_path):
+        download_dmp(admin)
+    elif redownload:
+        os.remove(raw_path)
+        download_dmp(admin)
+
+    df = pd.read_csv(raw_path, na_values=["-9998.8"])
+    df.dropna(axis="columns", how="all", inplace=True)
+    return df
+
+
+def process_dmp(
+    admin: AdminArgument = "ADM2",
+    redownload: bool = False,
+    start_dekad: int = 10,
+):
+    """Process DMP raw data
+
+    DMP raw data is received in a wide format
+    dataset that needs processing. This pivots
+    the data from wide to long format with
+    DMP dekadal observations for all years, while
+    also pivoting the mean values to allow for
+    calculation of biomasse anomaly.
+    """
+    df = load_dmp(admin, redownload)
+
+    # process mean and DMP separately since mean values
+    # are dekadal and DMP are year/dekadal
+    # keep ID column for working with multipolygon admin areas
+    df_mean = df.filter(regex="(^admin|^DMP_MEA|^ID)")
+    df_dmp = df.filter(regex="(^admin|^DMP_[0-9]+|^ID)")
+    admin_cols = [col for col in df_mean.columns if col.startswith("admin")]
+
+    # groupby to average out for the few cases where admin areas
+    # are not contiguous
+    df_mean_long = (
+        pd.wide_to_long(
+            df_mean,
+            i=admin_cols + ["IDBIOHYDRO"],
+            j="dekad",
+            stubnames="DMP_MEA",
+            sep="_",
+        )
+        .reset_index()
+        .drop(labels="IDBIOHYDRO", axis=1)
+        .groupby(admin_cols + ["dekad"])
+        .mean()
+        .reset_index()
+    )
+
+    # calculate anomaly for mean separate from
+    # observed to ensure unique dekadal values
+    df_mean_long["season_index"] = np.where(
+        df_mean_long["dekad"] >= start_dekad,
+        df_mean_long["dekad"] - start_dekad,
+        df_mean_long["dekad"] + (36 - start_dekad),
+    )
+    df_mean_long.sort_values(by=admin_cols + ["season_index"], inplace=True)
+    df_mean_long["biomasse_mean"] = df_mean_long.groupby(by=admin_cols)[
+        "DMP_MEA"
+    ].apply(lambda x: x.cumsum() * 365.25 / 36)
+
+    df_dmp_long = (
+        pd.wide_to_long(
+            df_dmp,
+            i=admin_cols + ["IDBIOHYDRO"],
+            j="time",
+            stubnames="DMP",
+            sep="_",
+        )
+        .reset_index()
+        .drop(labels="IDBIOHYDRO", axis=1)
+        .groupby(admin_cols + ["time"])
```

Comment:
you might want to check here first that there are no duplicated `[admin_cols]+["time"]` values? Since we saw this occur once? And now you are summing the values in case there is a duplicate, which I am doubtful whether that is what we should do. (especially since you are taking the mean after the groupby on `df_mean`

Reply:
Aha so this is the same issue with the polygons being split into two. Just make sure that the method after aggregation (I think it should be sum?) is the same with the dmp and mean data

Reply:
Yes, making sure they are all aligned! We take the mean in the first instance because we want to get a single measure of productivity for each unique area, and summing would be an overstimate. However, now we have a measure of productivity for each area so we want to sum it across our time period.

Reply:
@caldwellst I am not sure if I agree with this. I get the mean and the sum in general but when there is duplicate areas, I don't think it is correct. Maybe I am wrong but want to make sure we are not doing smth wrong. 

For example in Liberia we have the admin2 area with pcode `LR0402`. When I use the processed file (`biomasse_general_ADM2_dekad_10.csv`) and check the anomalies for that pcode, I see that they are always 180 or higher. If I understand the anomaly correctly the average anomaly should be 100 and thus it is not correct that it is always higher than 100. I would think the reason is that for the mean you take the mean over the two noncontigunous polygons while for the dekad-yearly DMP you take the sum. What is your take on this?

Reply:
Aha yes I see now. I think moving to a weighted average for the DMP values themselves makes sense then. I will run this and test if it's comparable. Thanks for the catch here!